### DISCUSSION TITLE
Difference between `depName` and `packageName`?

### DISCUSSION THREAD

**Discussion by byronmccollum** [renovatebot/renovate#14885] [Request Help] [answered]
I'm having some difficulty understanding the difference between `depName` and `packageName`, and when you would use one over the other? I didn't find much detail in the docs, except for these few lines...

From the `regexManager` docs:

> You must have either a `depName` capture group or a `depNameTemplate` config field
> You can optionally have a `packageName` capture group or a `packageNameTemplate` if it differs from `depName`

**Comment by rarkins** [comment, marked as answer]
depName is essentially the "display name" while packageName is the actual package name used to do the lookup.

If in doubt, just use depName. When packageName is blank, it defaults to depName's value. In most cases they're the same.

Sometimes though the name of the package is quite long, and you don't want that full name in commit messages and PR titles. So when you use both depName and packageName, depName is used in logs, commit messages and PRs while packageName is used to do the actual lookup from package registries

**Reply by byronmccollum** [reply]
Thank you!
